test/library/metrics/query.go

```diff
@@ -0,0 +1,107 @@
+package metrics
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	routeclient "github.com/openshift/client-go/route/clientset/versioned"
+	"github.com/prometheus/common/model"
+)
+
+type PrometheusClient struct {
+	// Host address of Prometheus public route
+	host string
+	// ServiceAccount bearer token to pass though OpenShift oauth proxy
+	token string
+}
+
+func NewPrometheusClient(client *kubernetes.Clientset, rc *routeclient.Clientset) (*PrometheusClient, error) {
+	_, err := client.CoreV1().Services("openshift-monitoring").Get("prometheus-k8s", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	route, err := rc.RouteV1().Routes("openshift-monitoring").Get("prometheus-k8s", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	host := route.Spec.Host
+	var bearerToken string
+	secrets, err := client.CoreV1().Secrets("openshift-monitoring").List(metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not list secrets in openshift-monitoring namespace")
+	}
+	for _, secret := range secrets.Items {
+		if secret.Type != corev1.SecretTypeServiceAccountToken {
+			continue
+		}
+		if !strings.HasPrefix(secret.Name, "prometheus-") {
+			continue
+		}
+		bearerToken = string(secret.Data[corev1.ServiceAccountTokenKey])
+		break
+	}
+	if len(bearerToken) == 0 {
+		return nil, fmt.Errorf("error getting bearer token from prometheus service account")
+	}
+	return &PrometheusClient{
+		host:  host,
+		token: bearerToken,
+	}, nil
+}
+
+type prometheusResponse struct {
+	Status string                 `json:"status"`
+	Data   prometheusResponseData `json:"data"`
+}
+
+type prometheusResponseData struct {
+	ResultType string       `json:"resultType"`
+	Result     model.Vector `json:"result"`
+}
+
+// Query returns metrics from external prometheus url or an error
+// ex: query := `ALERTS{alertstate="firing",alertname="Watchdog",severity="none"} == 1`
+// ex: query := `scheduler_scheduling_duration_seconds_sum`
+func (c *PrometheusClient) Query(query string) (model.Vector, error) {
+	var metrics model.Vector
+
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
cheating! what would it take to make this secure?

Reply:
I had this working in an earlier iteration, we'd need to include the CA (create configmap w/ annotation for service-ca-operator to create cert), then use the internal service url.  I still have that code somewhere locally - was steered in this direction following how cluster-monitoring folks configure their tests.

Reply:
@mfojtik Given that the monitoring team is also skipping verification in this test-only scenario, what is the value in other teams not doing the same? It's not like the security of the route's endpoints won't be validated by things like the console.

Reply:
@sallyom If verification is required, I think it would be preferable to verify the externally-exposed route rather than jumping through the hoops required to use the internal service url from tests running outside the cluster.